table/column.go

```diff
@@ -200,9 +210,28 @@ func CastValue(ctx sessionctx.Context, val types.Datum, col *model.ColumnInfo, r
 		truncateTrailingSpaces(&casted)
 	}
 
+	if col.Charset == charset.CharsetASCII {
+		if ctx.GetSessionVars().SkipASCIICheck {
```

Comment:
Please add a test for this variable, rest LGTM.

Reply:
Added, and this is the output in case that the jenkins test failed.
```
PASS: statement_context_test.go:32: testSuite1.TestStatementContext     0.055s
```